core/src/main/scala/org/apache/spark/serializer/JavaSerializer.scala

```diff
@@ -138,7 +142,7 @@ private[spark] class JavaSerializerInstance(
  */
 @DeveloperApi
 class JavaSerializer(conf: SparkConf) extends Serializer with Externalizable {
-  private var counterReset = conf.getInt("spark.serializer.objectStreamReset", 100)
+  private var counterReset = conf.getInt("spark.serializer.objectStreamReset", 3)
```

Comment:
A user would normally find this out when he / she hits OOME.

100, IMO, is big.
How about using 10 as the default ?